data/data/azure/vnet/vnet.tf

```diff
@@ -17,6 +17,8 @@ resource "azurerm_subnet" "master_subnet" {
   ] : cidr.value if cidr.include]
   virtual_network_name = local.virtual_network
```

Comment:
Since we are using the output of the created resource, we already depend on it... why do we need the explicit depend?

Reply:
I agree that you would /think/ that would handle the issue. Some googling around says otherwise. I agree this is a hack, which is why I've labeled the PR a WIP. Nobody seems to know for sure if this is a dependency problem within terraform, or a parallelism problem with Azure. One thing for sure, is that if you set parallelism=1, then all problems go away ;-) Here are some issues on github for reference:

https://github.com/terraform-providers/terraform-provider-azurerm/issues/3780
https://github.com/terraform-providers/terraform-provider-azurerm/issues/4285
https://github.com/terraform-providers/terraform-provider-azurerm/pull/3673

Reply:
+1

Reply:
We should also try to fix this upstream to make sure we don't need this depends on later